src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -984,6 +991,31 @@ public override string ToString()
             }
         }
 
+        /// <summary>
+        ///   Get a JsonElement which can be safely stored beyond the lifetime of the
+        ///   original JsonDocument.
+        /// </summary>
+        /// <returns>
+        ///   A JsonElement which can be safely stored beyond the lifetime of the
+        ///   original JsonDocument.
+        /// </returns>
+        /// <remarks>
+        ///   If this JsonElement is already detached (<see cref="IsDetached"/> ==
+        ///   <see langword="true"/>) and it is the root element of the
```

Comment:
Why does it have to be a root element (for the "no work" case)?

Reply:
> Why does it have to be a root element (for the "no work" case)?

If it does nothing then Detach() to hold 10k out of 10M keeps the 10M alive.  If it always copies then a large sub-object makes a large copy.

Since it felt like one of the reasons to use Detach was to "make the hold smaller" it seemed like "always" copy was better, but with the optimization that if it's already private immutable memory then no work needs to be done.